### PR TITLE
WIP, DO NOT MERGE

### PR DESCRIPTION
**Pull request by jbondpdx**
Changes to regsubst function for docs template work.

### CODE REVIEW

**Comment by jbondpdx** [comment]
@kris-bosland , here's a PR showing the template changes I'd like, on regsubst as the example. This is great progress, though!

File: source/puppet/latest/function.md [hunk 1]
```diff
@@ -3745,58 +3745,69 @@ $merged = $data.reduce( {} ) |$memo, $x| {
...
+Get the third octet from the node's IP address:
+
+```puppet
 $i3 = regsubst($ipaddress,'^(\\d+)\\.(\\d+)\\.(\\d+)\\.(\\d+)$','\\3')
 ```
-    * Return type(s): `Array[String]`, `String`. The result of the substitution. Result type is the same as for the target parameter.
-* `regsubst(Variant[Array[String],String] $target, Variant[Regexp,Type[Regexp]] $pattern, Variant[String,Hash[String,String]] $replacement, Optional[Pattern[/^G?$/]] $flags)`
-    * `target` --- The string or array of strings to operate on.  If an array, the replacement will be
+
+Return type(s): `Array[String]`, `String`. The result of the substitution. Result type is the same as for the target parameter.
```
Comment (jbondpdx) [inline source/puppet/latest/function.md]: The return type looks good, but should come before the example.

File: source/puppet/latest/function.md [hunk 2]
```diff
@@ -3745,58 +3745,43 @@ $merged = $data.reduce( {} ) |$memo, $x| {
...
+* `encoding` --- Optional. How to handle multibyte characters when compiling the regexp (must not be used when pattern is a
 precompiled regexp). A single-character string with the following values:
   - *N*         None
   - *E*         EUC
   - *S*         SJIS
   - *U*         UTF-8
-    `Get the third octet from the node's IP address:`
-      ```puppet
+
+Return type(s): `Array[String]`, `String`. The result of the substitution. Result type is the same as for the target parameter.
```
Comment (jbondpdx) [inline source/puppet/latest/function.md]: The return type should come before the examples.

File: source/puppet/latest/function.md [hunk 3]
```diff
@@ -3745,58 +3745,43 @@ $merged = $data.reduce( {} ) |$memo, $x| {
...
+
+Get the third octet from the node's IP address:
+
+```puppet
 $i3 = regsubst($ipaddress,'^(\\d+)\\.(\\d+)\\.(\\d+)\\.(\\d+)$','\\3')
 ```
-    * Return type(s): `Array[String]`, `String`. The result of the substitution. Result type is the same as for the target parameter.
```
Comment (jbondpdx) [inline source/puppet/latest/function.md]: All of this information is repeated before each dispatch call in regsubst.rb. So this is an error in the source, but if we delete the repeated information from the second dispatch call, does the second example still show up as it should?

File: source/puppet/latest/function.md [hunk 4]
```diff
@@ -3745,58 +3745,69 @@ $merged = $data.reduce( {} ) |$memo, $x| {
...
 
-* `regsubst(Variant[Array[String],String] $target, String $pattern, Variant[String,Hash[String,String]] $replacement, Optional[Optional[Pattern[/^[GEIM]*$/]]] $flags, Optional[Enum['N','E','S','U']] $encoding)`
-    * `target` --- The string or array of strings to operate on.  If an array, the replacement will be
+`regsubst(Variant[Array[String],String] $target, String $pattern, Variant[String,Hash[String,String]] $replacement, Optional[Optional[Pattern[/^[GEIM]*$/]]] $flags, Optional[Enum['N','E','S','U']] $encoding)`
+
+### Parameters
+
+* `target` --- The string or array of strings to operate on.  If an array, the replacement will be
 performed on each of the elements in the array, and the return value will be an array.
-    * `pattern` --- The regular expression matching the target string.  If you want it anchored at the start
```
[the remark this answers is not in the document]
Comment (kris-bosland) [reply source/puppet/latest/function.md]: *indented? :+1:

File: source/puppet/latest/function.md (hunk 3 above)
Comment (kris-bosland) [reply source/puppet/latest/function.md]: We should review what should be per-dispatch.

**Comment by kris-bosland** [comment]
@jbondpdx , I will update my PR to fix these issues and then ask you for comment there.  Thanks!

**Pull request closed without merging**